jerry-core/ecma/builtin-objects/ecma-builtin-date.c

```diff
@@ -367,13 +368,24 @@ ecma_builtin_date_parse_ISO_string_format (const lit_utf8_byte_t *date_str_curr_
           ecma_number_t timezone_offset = ecma_date_make_time (hours, minutes, ECMA_NUMBER_ZERO, ECMA_NUMBER_ZERO);
           time += is_timezone_sign_negative ? timezone_offset : -timezone_offset;
         }
+        else
+        {
+          is_time_in_local_time_zone = true;
+        }
       }
     }
 
     if (date_str_curr_p >= date_str_end_p)
     {
       ecma_number_t date = ecma_date_make_day (year, month - 1, day);
-      return ecma_date_make_date (date, time);
+
+      ecma_number_t result_date = ecma_date_make_date (date, time);
+      if (is_time_in_local_time_zone)
```

Comment:
`ecma_date_make_date` May return NaN, so I'd not pass it to `ecma_date_utc`.

Reply:
ecma_date_utc only subtracts the local time zone offset
and subtracting anything from NaN will be always NaN.

Or did I miss something?

Reply:
Indeed, but there is also a `localtime_r` calculation which can be skipped if we already know that the number is nan.

Reply:
I think parsing invalid dates is very a rare case and we should optimize for common cases.
If we did NaN check here, the rare case could be faster, but the common case would be slower.

Reply:
We are not dealing about heavy optimization. Only about to not perform operations which are unnecessary to do.

Reply:
I'm not trying to do any heavy optimization here. But your suggested NaN check here would obviously slow down the common case (valid date input) in order not to perform some extra operations in a very rare case (invalid date input). I still don't agree with you to optimize the code for rare cases with slowing down the common cases.

Otherwise I haven't seen any NaN check before ecma_date_utc calls in the code base.

Reply:
Convinced.